orttraining/orttraining/python/training/ortmodule/_custom_gradient_registry.py

```diff
@@ -125,3 +132,5 @@ def adaptive_avg_pool2d_gradient():
         (('ATenOp', 'com.microsoft'), ['GO(0)', 'I(0)'], [
          'GI(0)'], {'name': {'value': 'aten::_adaptive_avg_pool2d_backward', 'dtype': 'string'}}),
     ]
+
+CustomGradientRegistry.register_custom_stop_gradient_edges([0], 'com.microsoft', 'ATenOp', 'aten::multinomial', '')
```

Comment:
Vincent, please share your thoughts on using this approach to specific the non-differentiable edges. 

Another design option is to introduce an attribute '_differentiable_inputs', with value being the indexes of differentiable inputs.

Reply:
I actually think that for this specific case here, using registry_stop_edges is better than using attribute. Because this is a schema-level property. Using attribute is more meaningful if it's possible that different nodes of same Op type can have different attribute values. But for the case here the 0th input is always non-differentiable. Using attribute also requires extra work to set the attribute value when generating forward graph. It might be simple for current custom symbolic registration, but exporter may not happy to take this work in the future.

But do we really need to add such register_stop_edges for now? Actually we can stop to build gradient for this ATenOp when looking into the output type of this Op. It's only 1 output with INT64 dtype. We can always ignore such nodes. We can ignore nodes if none of the outputs is decimal type.

Reply:
I think adding the condition of not following non-decimal edges can be used to remove may ops like Shape, Size, GatherND etc from the STOP_GRADIENT_EDGES list. But since that condition alone is not enough, adding the same mechanism for Aten ops allows us to keep the gradient building logic consistent. Additionally, this allows us to have a mechanism for future aten ops where just the INT64 edge condition is not enough.
We can consider refactoring the gradient builder to add this rule of INT64 in a separate PR if necessary.